client/my-sites/theme/controller.jsx

```diff
@@ -75,9 +76,11 @@ export function details( context, next ) {
 
 export function notFoundError( err, context, next ) {
 	context.layout = (
-		<ReduxProvider store={ context.store }>
-			<LayoutLoggedOut primary={ <ThemeNotFoundError /> } />
-		</ReduxProvider>
+		<RouteProvider currentSection={ context.section } currentQuery={ context.query }>
```

Comment:
I guess the original way looks weird. Maybe we just need to set the `context.primary` to `<ThemeNotFoundError />`, and that's enough as we use `RouteProvider` in any other entry point. WDYT?

Reply:
Oh, that's a good point. Although I'm curious why `<ThemeNotFoundError/>` is forced to render with the logged out layout 🤔

Reply:
hmmm, that's a great point 🙃